checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -471,16 +473,33 @@ public Void visitInstanceOf(InstanceOfTree tree, Void p) {
             // Handle them properly.
             return null;
         }
+
+        List<AnnotationMirror> annotations = new ArrayList<>();
+
         if (refTypeTree.getKind() == Tree.Kind.ANNOTATED_TYPE) {
-            List<? extends AnnotationMirror> annotations =
-                    TreeUtils.annotationsFromTree((AnnotatedTypeTree) refTypeTree);
-            if (AnnotationUtils.containsSame(annotations, NULLABLE)) {
-                checker.reportError(tree, "instanceof.nullable");
-            }
-            if (AnnotationUtils.containsSame(annotations, NONNULL)) {
-                checker.reportWarning(tree, "instanceof.nonnull.redundant");
+            annotations.addAll(TreeUtils.annotationsFromTree((AnnotatedTypeTree) refTypeTree));
+        }
+
+        Tree patternTree = TreeUtilsAfterJava11.InstanceOfUtils.getPattern(tree);
+        if (patternTree != null) {
+            if (TreeUtils.isBindingPatternTree(patternTree)) {
```

Comment:
Can these two be combined?
```suggestion
        if (patternTree != null && TreeUtils.isBindingPatternTree(patternTree)) {
```